src/gtktest.rs

```diff
@@ -3,295 +3,182 @@
+            println!("spin_button_input: \"{}\"", text);
+            match text.parse::<f64>() {
+                Ok(value) if value >= 90. => {
+                    println!("circular right");
+                    Some(Ok(10.))
+                }
+                Ok(value) if value <= 10. => {
+                    println!("circular left");
+                    Some(Ok(90.))
+                }
+                Ok(value) => Some(Ok(value)),
+                Err(_) => Some(Err(())),
+            }
+        });
+
+        let button: Button = builder.get_object("button").unwrap();
+        button.connect_clicked(clone!(window, entry => move |_| {
+            let dialog = Dialog::new_with_buttons(Some("Hello!"), Some(&window), gtk::DIALOG_MODAL,
+                &[("No", 0), ("Yes", 1), ("Yes!", 2)]);
+
+            let ret = dialog.run();
+
+            dialog.destroy();
+
+            entry.set_text(&format!("Clicked {}", ret));
+        }));
+
+        let button_font: Button = builder.get_object("button_font").unwrap();
+        button_font.connect_clicked(clone!(window => move |_| {
+            let dialog = FontChooserDialog::new(Some("Font chooser test"), Some(&window));
+
+            dialog.run();
+            dialog.destroy();
+        }));
+
+        let button_recent: Button = builder.get_object("button_recent").unwrap();
+        button_recent.connect_clicked(clone!(window => move |_| {
+            let dialog = RecentChooserDialog::new(Some("Recent chooser test"), Some(&window));
+            dialog.add_buttons(&[
+                ("Ok", ResponseType::Ok.into()),
+                ("Cancel", ResponseType::Cancel.into())
+            ]);
+
+            dialog.run();
+            dialog.destroy();
+        }));
+
+        let file_button: Button = builder.get_object("file_button").unwrap();
+        file_button.connect_clicked(clone!(window => move |_| {
+            //entry.set_text("Clicked!");
+            let dialog = FileChooserDialog::new(Some("Choose a file"), Some(&window),
+                                                FileChooserAction::Open);
+            dialog.add_buttons(&[
+                ("Open", ResponseType::Ok.into()),
+                ("Cancel", ResponseType::Cancel.into())
+            ]);
+
+            dialog.set_select_multiple(true);
+            dialog.run();
+            let files = dialog.get_filenames();
+            dialog.destroy();
+
+            println!("Files: {:?}", files);
+        }));
+
+        let app_button: Button = builder.get_object("app_button").unwrap();
+        app_button.connect_clicked(clone!(window => move |_| {
+            //entry.set_text("Clicked!");
+            let dialog = AppChooserDialog::new_for_content_type(Some(&window), gtk::DIALOG_MODAL,
+                "sh");
+
+            dialog.run();
+            dialog.destroy();
+        }));
+
+        let button_about: Button = builder.get_object("button_about").unwrap();
+        button_about.connect_clicked(move |x| {
```

Comment:
It comes from the fact that Glade creates only one instance of it, so we need to show/hide instead of destroy. I added a comment in the source code to explain it.